pixart.js

```diff
@@ -1 +1,46 @@
+$(document).ready(function() {
+  var colorSwatch = []; //instantiate dummy array accessible to all functions
 
+  //set brush color helper function
+  function setBrushColor() {
+    var brushColor = $("#color-field").val(); //get the current brush color (from the .brush div)
+    $(".brush").css('background', brushColor); //set the brush to the new value
+    addToColorSwatch(brushColor); //add that color to the color swatch
```

Comment:
Not sure if this will solve your problem, but what if you did something along the lines of this...

* Create a variable `colorToBeAddedToSwatch`
* `colorToBeAddedToSwatch` is set to the new current color when you click "Set Color"
* `colorToBeAddedToSwatch` is added to the swatch the **next** time the user clicks "Set Color"

So you're basically creating a staging area for the swatch. The current color is not added to the swatch until it is replaced by a new current color.